WordPress/src/test/java/org/wordpress/android/ui/posts/FeaturedImageHelperTest.kt

```diff
@@ -0,0 +1,297 @@
+        // Act
+        val failedFeaturedImage = featuredImageHelper.getFailedFeaturedImageUpload(mock())
+        // Assert
+        assertThat(failedFeaturedImage).isNotNull
+    }
+
+    @Test
+    fun `retryFeaturedImageUpload sets MediaUploadState to QUEUED`() {
+        // Arrange
+        whenever(uploadStore.getFailedMediaForPost(anyOrNull())).thenReturn(setOf(createMediaModel(true)))
+        // Act
+        val mediaModel = featuredImageHelper.retryFeaturedImageUpload(mock(), mock())
+        // Assert
+        assertThat(MediaUploadState.fromString(mediaModel!!.uploadState)).isEqualTo(MediaUploadState.QUEUED)
+    }
+
+    @Test
+    fun `retryFeaturedImageUpload dispatches updateMediaAction`() {
+        // Arrange
+        whenever(uploadStore.getFailedMediaForPost(anyOrNull())).thenReturn(setOf(createMediaModel(true)))
+        val captor: KArgumentCaptor<Action<MediaModel>> = argumentCaptor()
+        // Act
+        val mediaModel = featuredImageHelper.retryFeaturedImageUpload(mock(), mock())
+        // Assert
+        verify(dispatcher).dispatch(captor.capture())
+        assertThat(captor.firstValue).matches { action -> action.type == MediaAction.UPDATE_MEDIA }
+        assertThat(captor.firstValue).matches { action -> action.payload == mediaModel }
+    }
+
+    @Test
+    fun `retryFeaturedImageUpload starts the UploadService`() {
+        // Arrange
+        whenever(uploadStore.getFailedMediaForPost(anyOrNull())).thenReturn(setOf(createMediaModel(true)))
+        // Act
+        val mediaModel = featuredImageHelper.retryFeaturedImageUpload(mock(), mock())
+        // Assert
+        verify(uploadServiceFacade).uploadMedia(argThat { this[0] == mediaModel })
+    }
+
+    @Test
+    fun `queueFeaturedImageForUpload marks the MediaModel as featuredImage`() {
+        // Arrange
+        val mediaModel = createMediaModel(markedLocallyAsFeatured = false)
+        whenever(fluxCUtilsWrapper.mediaModelFromLocalUri(anyOrNull(), anyOrNull(), anyInt())).thenReturn(mediaModel)
+        // Act
+        featuredImageHelper.queueFeaturedImageForUpload(0, createSiteModel(), mock(), "")
+        // Assert
+        assertThat(mediaModel.markedLocallyAsFeatured).isTrue()
+    }
+
+    @Test
+    fun `queueFeaturedImageForUpload returns false when the uri can't be converted to MediaModel`() {
+        // Arrange
+        whenever(fluxCUtilsWrapper.mediaModelFromLocalUri(anyOrNull(), anyOrNull(), anyInt())).thenReturn(null)
+        // Act
+        val result = featuredImageHelper.queueFeaturedImageForUpload(0, createSiteModel(), mock(), "")
+        // Assert
+        assertThat(result).isFalse()
+    }
+
+    @Test
+    fun `queueFeaturedImageForUpload returns false when the provided localPostId is empty`() {
+        // Arrange
+        whenever(fluxCUtilsWrapper.mediaModelFromLocalUri(anyOrNull(), anyOrNull(), anyInt()))
+                .thenReturn(createMediaModel(markedLocallyAsFeatured = false))
```

Comment:
Extremely super minor np: I would consider to align on using or not the named argument for markedLocallyAsFeatured parameters throughout the file.

Reply:
Good call! ;) Fixed in 1c2625dcff70cc9551838f796fec36eef9aff073